.github/actions/pub_publish/action.yaml

```diff
@@ -0,0 +1,42 @@
+name: Publish to pub.dev
+description: Publish a package to pub.dev
+
+inputs:
+  working_directory:
+    description: Directory within the repository where the package is located (defaults to root).
+    required: false
+
+runs:
+  using: composite
+  steps:
+    - name: 📚 Git Checkout
+      uses: actions/checkout@v4
+
+    - name: 🎯 Setup Dart
+      uses: dart-lang/setup-dart@v1
+
+    - name: 🪪 Get Token
+      uses: actions/github-script@v6
+      with:
+        script: |
+          const pub_token = await core.getIDToken('https://pub.dev')
```

Comment:
Oh, interesting

Reply:
yup this way we don't need to maintain pub tokens ourselves 😄